src/components/KCopy/KCopy.vue

```diff
@@ -0,0 +1,274 @@
+      position-fixed
+    >
+      <div
+        :class="[
+          'content-container',
+          truncate ? 'truncated-content' : 'non-truncated-content',
+          monospace ? 'monospace' : null
+        ]"
+        :style="widthStyle"
+      >
+        {{ contentFormat }}
+      </div>
+    </KTooltip>
+
+    <KTooltip
+      class="content-icon-wrapper"
+      :label="`${props.badgeText}` ? iconTitle : tooltipText"
+      max-width="500px"
+      placement="bottomStart"
+      position-fixed
+    >
+      <KClipboardProvider v-slot="{ copyToClipboard }">
+        <span
+          data-testid="copy-to-clipboard"
+          role="button"
+          tabindex="0"
+          @click.stop="copyIdToClipboard(copyToClipboard)"
+        >
+          <KIcon
+            class="content-icon"
+            :hide-title="!!copyTooltip || undefined"
+            icon="copy"
+            size="16"
+            :title="iconTitle"
+          />
+        </span>
+      </KClipboardProvider>
+    </KTooltip>
+  </div>
+
+  <div
+    v-else
+    class="k-copy-badge"
+  >
+    <span
+      v-if="isBadge"
+      class="copy-badge-text"
+    >
+      {{ badgeText }}
+    </span>
+    <KBadge
+      v-if="isBadge"
+      appearance="custom"
+      background-color="#f9fafb"
+      border-color="#d7d8fe"
+      class="id-badge"
+      color="#473cfb"
+      is-bordered
+      shape="rectangular"
+    >
+      <KCopy
+        :badge-text="badgeText"
+        class="id-badge-content"
+        :content="content"
+        :content-tooltip="contentTooltip"
+        :copy-tooltip="copyTooltip"
+        :monospace="monospace"
+        :success-tooltip="successTooltip"
+        :truncate="truncate"
+        :truncation-limit="truncationLimit"
+      />
+    </KBadge>
+  </div>
+</template>
+
+<script setup lang="ts">
+import type { PropType } from 'vue'
+import { computed, ref } from 'vue'
+import KBadge from '@/components/KBadge/KBadge.vue'
+import KIcon from '@/components/KIcon/KIcon.vue'
+import KClipboardProvider from '@/components/KClipboardProvider'
+import KTooltip from '@/components/KTooltip/KTooltip.vue'
+
+const props = defineProps({
+  // text displayed before the copyable content when `isBadge` is true
+  badgeText: {
+    type: String,
+    default: '',
+  },
+  // the copyable text
+  content: {
+    type: String,
+    required: true,
+  },
+  // tooltip text displayed on hover over the `content`
+  contentTooltip: {
+    type: String,
+    default: '',
+  },
+  // tooltip text displayed on hover over copy button
+  copyTooltip: {
+    type: String,
+    default: 'Copy',
+  },
+  // formatting for copyable content (default, hidden, redacted, deleted)
+  format: {
+    type: String as PropType<'default' | 'hidden' | 'redacted' | 'deleted'>,
+    required: false,
+    default: 'default',
+    validator: (val: string) => ['default', 'hidden', 'redacted', 'deleted'].includes(val),
+  },
+  // whether or not to display as a badge
+  isBadge: {
```

Comment:
issue: just change to `badge` so we follow our "no prefixed props" standard
```suggestion
  badge: {
```